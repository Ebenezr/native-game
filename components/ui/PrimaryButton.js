import { Pressable, View, Text, StyleSheet } from 'react-native';
import Colors from '../../constants/colors';

function PrimaryButton({ children, onPress }) {
  return (
    <View style={styles.buttonContainer}>
      <Pressable
        style={({ pressed }) =>
          pressed ? [styles.button, styles.pressed] : styles.button
        }
        onPress={onPress}
        android_ripple={{ color: '#A0567A' }}
      >
        <Text style={styles.buttonText}>{children}</Text>
      </Pressable>
    </View>
  );
}
export default PrimaryButton;

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.primary50,
    paddingVertical: 8,
    paddingHorizontal: 16,
    elevation: 5,
  },
  buttonText: {
    color: 'white',
    textAlign: 'center',
  },
  buttonContainer: {
    margin: 8,
    borderRadius: 50,
    overflow: 'hidden',
    flex: 1,
  },
  pressed: {
    opacity: 0.75,
  },
});
