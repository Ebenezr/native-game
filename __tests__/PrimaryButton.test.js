import renderer from 'react-test-renderer';
import PrimaryButton from '../components/ui/PrimaryButton';

describe('PrimaryButton', () => {
  it('renders Button correctly', () => {
    const tree = renderer.create(<PrimaryButton title='Test Title' />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
