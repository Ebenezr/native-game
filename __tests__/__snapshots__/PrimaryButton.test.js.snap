// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#72063c",
      "borderRadius": 10,
      "elevation": 5,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginTop": 40,
      "padding": 20,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 6,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Test Title
  </Text>
  Test Children
</View>
`;

exports[`PrimaryButton renders Button correctly 1`] = `
<View
  style={
    {
      "borderRadius": 50,
      "flex": 1,
      "margin": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#A0567A",
        "elevation": 5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "open-sans-bold",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`PrimaryButton renders Card correctly 1`] = `
<View
  style={
    {
      "borderRadius": 50,
      "flex": 1,
      "margin": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#A0567A",
        "elevation": 5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "open-sans-bold",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
